go/tools/builders/compilepkg.go

```diff
@@ -612,6 +603,15 @@ func runNogo(ctx context.Context, workDir string, nogoPath string, srcs []string
 	return nil
 }
 
+func appendToArchive(goenv *env, outPath string, objFiles []string) error {
+	// Use abs to work around long path issues on Windows.
+	// TODO(jayconrod): copy cmd/internal/archive and use that instead of
+	// shelling out to cmd/pack.
```

Comment:
Nit: I don’t think we should consider this TODO as it will increase the maintenance effort for rules_go. Shelling out help us ensure compatibility against upstream Go SDK.